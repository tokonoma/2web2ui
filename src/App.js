import React from 'react';
import { AuthenticationGate, SuspensionAlerts } from 'src/components/auth';
import { CookieConsent, GlobalAlertWrapper, BoomerangBanner, SiftScience } from 'src/components';
import VisualWebsiteOptimizer from './components/vwo/VisualWebsiteOptimizer';
import Segment from './components/segment/Segment';
import Support from 'src/components/support/Support';
import GoogleTagManager from 'src/components/googleTagManager/GoogleTagManager';
import Pendo from 'src/components/pendo/Pendo';
import Layout from 'src/components/layout/Layout';
import AppRoutes from 'src/components/appRoutes';
import { HibanaBanner, HibanaToggle, HibanaDescription } from 'src/components/hibana';
import GlobalBanner from 'src/context/GlobalBanner';

import config from 'src/config';

import { BrowserRouter } from 'react-router-dom';

const App = ({ RouterComponent = BrowserRouter }) => (
  <RouterComponent>
    <div>
      {config.siftScience && <SiftScience config={config.siftScience} />}
      <BoomerangBanner />
      {config.gtmId && <GoogleTagManager id={config.gtmId} />}
      <Pendo />
      <VisualWebsiteOptimizer />
      <Segment />
      <AuthenticationGate />
      <SuspensionAlerts />
      <CookieConsent />
      <GlobalBanner>
        <Layout>
          <AppRoutes />
        </Layout>
      </GlobalBanner>
      <Support />
      <GlobalAlertWrapper />

      <HibanaBanner>
        <HibanaDescription />
        <HibanaToggle />
      </HibanaBanner>
    </div>
  </RouterComponent>
);

export default App;
