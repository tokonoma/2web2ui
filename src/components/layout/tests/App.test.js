import React from 'react';
import { App } from '../App';
import { shallow } from 'enzyme';
import TestApp from 'src/__testHelpers__/TestApp';

describe('Component: App Layout', () => {
  it('should render correctly', () => {
    expect(
      shallow(
        <TestApp>
          <App>
            <h1>My cool af children</h1>
          </App>
        </TestApp>,
      ),
    ).toMatchSnapshot();
  });
});
