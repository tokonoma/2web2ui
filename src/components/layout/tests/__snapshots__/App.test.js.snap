// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: App Layout should render correctly 1`] = `
<Providers
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <MemoryRouter>
    <App>
      <h1>
        My cool af children
      </h1>
    </App>
  </MemoryRouter>
</Providers>
`;
